pkg/sql/logictest/testdata/logic_test/temp_table

```diff
@@ -241,3 +241,26 @@ query I
 SELECT * FROM regression_47030
 ----
 2
+
+# renaming a temp table should not move it to the public schema.
+subtest regression_48233
+
+statement ok
+CREATE TEMP TABLE regression_48233(a int)
+
+statement ok
+ALTER TABLE regression_48233 RENAME TO reg_48233
+
+query IITI rowsort
+SELECT * FROM system.namespace WHERE name LIKE '%48233'
+----
+50  65  reg_48233  79
+
+statement error cannot convert a temporary table to a persistent table during renames
+ALTER TABLE reg_48233 RENAME TO public.reg_48233
+
+statement ok
+CREATE TABLE persistent_48233(a int)
+
+statement error schema cannot be modified: "pg_temp"
```

Comment:
this isn't a great error message, but ok for this PR.

looks like the earlier code stops us from triggering this code path based on https://github.com/cockroachdb/cockroach/blob/c7fadb7dd88addaea0d7677aabd3f45b71cf7fbb/pkg/sql/catalog/resolver/resolver.go#L219... i'm starting to think we should actually fix this by changing the sql.y to only take in one table name, as postgres treats `schema_name.table_name` as a syntax error. we can do that in a separate PR as we should backport this to 20.1. 

schema changes should use the `SET SCHEMA` syntax instead. feel free to file a ticket.

i think we can delete the
```
isNewSchemaTemp, _, err := temporarySchemaSessionID(newTn.Schema())
	if err != nil {
		return err
	}

	if newTn.ExplicitSchema && !isNewSchemaTemp && tableDesc.Temporary {
		return pgerror.New(
			pgcode.FeatureNotSupported,
			"cannot convert a temporary table to a persistent table during renames",
		)
	}
```
block if we use the SET SCHEMA syntax and only let this take in a table name.

Reply:
It's probably better to go down this route, but it could break existing scripts. Agreed, rename table shouldn't have ever taken a qualification...